torch/backends/cudnn/__init__.py

```diff
@@ -37,8 +37,8 @@ def _init():
                 cudnn_compatible = runtime_minor >= compile_minor
             if not cudnn_compatible:
                 raise RuntimeError(
-                    'cuDNN version incompatibility: PyTorch was compiled against {} '
-                    'but linked against {}'.format(compile_version, runtime_version))
+                    'cuDNN version incompatibility: PyTorch was compiled  against {} '
+                    'but found runtime version {}. Please use cudnn version provided with pytorch.'.format(compile_version, runtime_version))
```

Comment:
nit: I'd add some more details:
"cuDNN version incompatibility: PyTorch was compiled against {} but found runtime version {}. PyTorch already comes bundled with cuDNN. One option to resolving this error is to ensure PyTorch can find the bundled cuDNN; one possibility is that there is a conflicting cuDNN in LD_LIBRARY_PATH"

Reply:
could PyTorch somehow print the actually-loaded library file path? (maybe by examining its own proc filesystem and vmaps? or using some wrapper like psutil for this). This may help debugging

Reply:
So probably can be done by reading `f"/proc/{os.getpid()}/maps"` (on Unix machines) and filtering cudnn .so files